src/apps/ociochecklut/main.cpp

```diff
@@ -357,59 +363,140 @@ int main (int argc, const char* argv[])
-                try
-                {
-                    proc.apply(pixel);
-                }
-                catch (const OCIO::Exception & e)
-                {
-                    std::cerr << "ERROR: Processing pixel: " << e.what() << std::endl;
-                    return 1;
-                }
-                catch (...)
-                {
-                    std::cerr << "ERROR: Unknown error encountered while processing pixel." << std::endl;
-                    return 1;
-                }
-
-                // Print to string so that in & out values can be aligned if needed.
 
-                std::vector<std::string> out;
-                ToString(out, pixel, 0, comp);
-
-                if (verbose)
+                if (stepInfo)
                 {
-                    std::vector<std::string> in;
-                    ToString(in, input, curPix, comp);
-
-                    std::cout << "Input  [R G B";
-                    if (comp == 4)
+                    // Process each step in a multi - transform LUT
+                    try
                     {
-                        std::cout << " A";
+                        // Create GroupTransform so that each can be processed one at a time. 
+                        auto processor = config->getProcessor(t);
+                        auto transform = processor->createGroupTransform();
+                        std::vector<float> inputPixel = pixel;
+                        std::vector<float> outputPixel = pixel;
+                        const auto numTransforms = transform->getNumTransforms();
+                        
+                        std::cout << std::endl;
+
+                        for (int i = 0; i < numTransforms; ++i)
+                        {
+                            auto transformStep = transform->getTransform(i);
+                            auto processorStep = config->getProcessor(transformStep);
+
+                            if (usegpu || usegpuLegacy)
+                            {
+                                proc.setGPU(usegpuLegacy ? processorStep->getOptimizedLegacyGPUProcessor(OCIO::OPTIMIZATION_DEFAULT, 32)
+                                    : processorStep->getDefaultGPUProcessor());
+                            }
+                            else
+                            {
+                                proc.setCPU(processorStep->getDefaultCPUProcessor());
+                            }
+                            
+                            // Process the pixel
+                            proc.apply(outputPixel);
+
+                            // Print the input/output pixel
+                            std::vector<std::string> in;
+                            ToString(in, inputPixel, 0, comp);
+
+                            std::vector<std::string> out;
+                            ToString(out, outputPixel, 0, comp);
+
+                            std::cout << "Transform " << i << ": " << std::endl;
```

Comment:
Rather than printing "Transform i" here, it would be more helpful to print a description of the transform:
`std::cout << "\n" << *(transform->getTransform(i)) << std::endl;`

And then please revert the change to line 275.

Reply:
Sure no problem @doug-walker , I made the changes.